src/FAI3_backend/src/model.rs

```diff
@@ -1,12 +1,14 @@
 use crate::{
     check_cycles_before_action, is_owner, only_admin, AverageMetrics, DataPoint, Metrics, Model,
-    ModelDetails, MODELS, NEXT_MODEL_ID,
+    ModelDetails, MODELS, NEXT_MODEL_ID
 };
 use candid::Principal;
 use std::vec;
+use crate::types::get_classifier_model_data;
+use crate::types::{ModelType, ClassifierModelData, LLMModelData};
 
 #[ic_cdk::update]
-pub fn add_model(model_name: String, model_details: ModelDetails) -> u128 {
+pub fn add_classifier_model(model_name: String, model_details: ModelDetails) -> u128 {
```

Comment:
It still has specific add_model methods for every tiype (add_classifier_model and add_llm_model).